test/test_get_soup.py

```diff
@@ -0,0 +1,40 @@
+import requests
+import mock
+import pytest
+import sys
+
+sys.modules["config"] = mock.MagicMock()
+from amz_review_scraper.get_soup import boil_soup
+
+
+def mocked_requests_get(*args, **kwargs):
+    class MockResponse:
+        def __init__(self, json_data, text, status_code):
+            self.json_data = json_data
+            self.text = text
+            self.status_code = status_code
+
+        def json(self):
+            return self.json_data
+
+    print("args: " + args[0])
+    if args[0] == "https://www.amazon.com/gp/product/B111111111":
+        return MockResponse({"key1": "value1"}, "sample text", 404)
+    elif args[0] == "https://www.amazon.com/gp/product/B07HJXVHSS":
+        return MockResponse({"key2": "value2"}, "sample text", 200)
+
+    return MockResponse(None, 404)
+
+
+def test_boil_soup():
+    with mock.patch("requests.get", side_effect=mocked_requests_get):
+        json_data1 = boil_soup(
+            url="https://www.amazon.com/gp/product/B07HJXVHSS", asin="B07HJXVHSS"
+        )
+    with mock.patch("requests.get", side_effect=mocked_requests_get):
+        json_data2 = boil_soup(
+            url="https://www.amazon.com/gp/product/B111111111", asin="B07HJXVHSS"
+        )
+
+    assert json_data1.status_code == None
```

Comment:
Some context might good here.

Like why did we want `json_data1` to be None, but not `json_data2`.

A good practice in writing tests, and watching them fail, and trying to image what someone else, or you later one would know from the failure.

For instance, if this failed, it would say something like `test_boil_soup failed json_data1.status_code is Not None`,

Which might not be that useful.

This is all made up.
But it would be like `def test_boil_soup_finds_product_if_it_exists` and then maybe rename the json_data variable, to be like `invalid_asin_result`.